src/Components/Endpoints/src/RazorComponentEndpoint.cs

```diff
@@ -0,0 +1,18 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Components.Endpoints;
+
+internal static class RazorComponentEndpoint
+{
+    public static RequestDelegate CreateRouteDelegate(Type componentType)
+    {
+        return httpContext =>
+        {
+            var result = new RazorComponentResult(componentType);
```

Comment:
Looks like the `RazorComponentResult` is not going to be modified so it could be moved out of the delegate?

Reply:
Can you clarify what benefit we'd get from that?

Reply:
While that's technically true right now, it will stop being true once we start passing extra information from the `HttpContext` into the result (e.g., for parameters).

Reply:
Also, result instances are not generally expected to be reused across different requests.

Reply:
BTW your comment made me think of restructuring this for more clarity about where this is going next, so thanks!